### ISSUE TITLE
Flyway parsing bug: unable to decrease block depth below 0 in version 6.3.0

### ISSUE THREAD

**Issue by rafael-trevisan** [flyway/flyway#2715]
###### Which version and edition of Flyway are you using?
6.3.0 Pro

###### If this is not the latest version, can you reproduce the issue with the latest one as well?
(Many bugs are fixed in newer releases and upgrading will often resolve the issue)
In v6.2.4 flyway was removing the ";" from my package spec. I upgraded to 6.3.0 and now it raises an error on the line that has the final ";"

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)
Command-line

###### Which database are you using (type & version)?
Oracle 18

###### Which operating system are you using?
MacOS

###### What did you do?
(Please include the content causing the issue, any relevant configuration settings, the SQL statement that failed (if relevant) and the command you ran.)
In v6.2.4 I was having an issue where Flyway was removing the last ";" from my package spec causing them to be invalid. I upgraded to 6.3.0 hoping the issue could go away but now Flyway raises an exception on the exact line that has the last ";"

Example of the code:
```
create or replace package package_a authid current_user as

  gc_a constant varchar2(1024)   := 'bla';
  g_b number;
  
  procedure x (
    some_param number
  );
end package_a;

**Comment by rafael-trevisan** [comment]
I downgraded to version 6.2.3 and the issue is gone. I'll stick with this version for the time being.

**Comment by rafael-trevisan** [comment]
I downgraded to version 6.2.0 and the issue is gone.

**Comment by Lyeeedar** [comment]
Fixed in c2297307f4bc056ea29a5e4d32f7eab8f39c81a4
